src/hooks/useElementWidth.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+
+const useElementWidth = (
+  ref: React.RefObject<HTMLElement | HTMLButtonElement | HTMLInputElement>
+): number => {
+  const [width, setWidth] = useState<number>(0);
+
+  useEffect(() => {
+    const updateWidth = () => {
+      const width = ref.current ? ref.current.offsetWidth : 0;
+      setWidth(width);
+    };
+
+    updateWidth();
+    window.addEventListener('resize', updateWidth);
+    return () => {
+      window.removeEventListener('resize', updateWidth);
+    };
+  }, [ref]);
```

Comment:
In theory refs should not be added in the dependency array. Is ESLint complaining to you?

```suggestion
  }, []);
```

Reply:
yes, eslint is compaining